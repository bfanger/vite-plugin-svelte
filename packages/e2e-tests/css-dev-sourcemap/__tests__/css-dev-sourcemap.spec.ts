import { browserLogs, getColor, getText, isBuild } from '~utils';

test('should not have failed requests', async () => {
	browserLogs.forEach((msg) => {
		expect(msg).not.toMatch('404');
	});
});

test('should apply css compiled from scss', async () => {
	expect(await getText('#test')).toBe('red');
	expect(await getColor('#test')).toBe('red');
});

if (!isBuild) {
	test('should generate sourcemap', async () => {
		expect(await getText('style')).toMatchSnapshot();
	});
}
