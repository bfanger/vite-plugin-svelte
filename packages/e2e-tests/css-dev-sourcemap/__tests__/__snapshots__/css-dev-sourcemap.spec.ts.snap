// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should generate sourcemap 1`] = `
"#test.s-XsEmFtvddWTw{color:red}.s-XsEmFtvddWTw{}
/*# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJmaWxlIjoiQXBwLnN2ZWx0ZSIsIm1hcHBpbmdzIjoiQUFJRSxvQkFBQSxDQUNDLEtBQUEsQ0FBQSIsIm5hbWVzIjpbXSwic291cmNlcyI6WyJBcHAuc3ZlbHRlIl0sInNvdXJjZXNDb250ZW50IjpbIjxkaXYgaWQ9XCJ0ZXN0XCI+cmVkPC9kaXY+XG5cbjxzdHlsZSBsYW5nPVwic2Nzc1wiPlxuXHQjdGVzdCB7XG5cdFx0JiB7XG5cdFx0XHRjb2xvcjogcmVkO1xuXHRcdH1cblx0fVxuPC9zdHlsZT5cbiJdfQ== */"
`;
